subprojects/launcher/src/main/java/org/gradle/launcher/daemon/server/health/LowNonHeapDaemonExpirationStrategy.java

```diff
@@ -27,7 +27,7 @@
     private final DaemonMemoryStatus status;
     private static final Logger LOG = Logging.getLogger(LowNonHeapDaemonExpirationStrategy.class);
 
-    public static final String EXPIRATION_REASON = "after running out of JVM memory";
+    public static final String EXPIRATION_REASON = "after running out of JVM metaspace";
```

Comment:
The previous messages were misleading, as they were the same for both heap and metaspace.

Reply:
If you look back through the history, this message was originally more explicit around what was happening but then there was an effort to make the messages simpler and the argument was that it's all jvm memory and the user doesn't care about the details.  I tend to agree that we should be more informational here and that we don't gain much (anything) by lumping this in with heap space issues.